src/NSubstitute/Core/CallSpecification.cs

```diff
@@ -123,7 +123,9 @@ public IEnumerable<ArgumentMatchInfo> NonMatchingArguments(ICall call)
 
     public override string ToString()
     {
-        var argSpecsAsStrings = _argumentSpecifications.Select(x => x.ToString() ?? string.Empty).ToArray();
+        var argSpecsAsStrings = _argumentSpecifications.Select(x =>
+            (x as IDescribeSpecification)?.DescribeSpecification() ?? x.ToString() ?? string.Empty
```

Comment:
Nice pick up 👍 

I have made the `IDescribeSpecification` value be used when implemented, with the minor tweak that null will be replaced with `string.Empty`. This matches the documented behaviour of `IDescribeNonMatches`. If operating in `#nullable enable` section this shouldn't matter; if operating without then we won't be using unexpectedly null values.

Happy to revise this decision based if you prefer it to use `null`.

Reply:
Personally, I'm not a fan of having `null`-ness control the behavior of code. (especially with types like `string` and `Enumerable` types), and given the purpose of the interface I don't see the point of users implementing it and returning `null` anyway so the specs are fine like this to me. Since we can't enforce this contract on user implementations, I agree it's the right thing to replace possible `null` returns with `String.Empty` to eliminate the impact of them.